### PR TITLE
Updating User auth from talk today; see the TODOs for considerations

### PR DESCRIPTION
**Pull request by jad-b**
Didn't make it all the way through, but I wanted to lay out the bigger ideas
- UserAuth username, user metadata, and all auth-related work
- UserPII holds all sensitive information that directly relates to a real person
- sessions.go is gone; auth tokens basically represent sessions

I think UserAuth could stand to be split into UserMeta, but I don't really care right now. What probably makes the most sense is whatever the most-common use-case for DB lookups will be (minimizing JOINs, etc.)

### CODE REVIEW

File: users/pii.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+// UserPIISQL is SQL for creating the table.
+const UserPIISQL = `
+CREATE TABLE auth.UserPII (
+  "id" numeric(5,2) NOT NULL
+  "username" text NOT NULL UNIQUE
+  "first_name" text NOT NULL
+  "last_name" text NOT NULL
+  "email" text NOT NULL UNIQUE
+  "enabled" bool
```
Comment (moogar0880) [inline users/pii.go]: Looks like the PII struct doesn't actually have any of the fields below `email`

File: users/pii.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+);`
+
+// UserPII contains sensitive user information which could identify them in the
+// real-world. Most of the time, you should only be accessing this if you need
+// to display the user's profile or send them an email. Or swag.
+//
+// PII (Personally Identifiable Information) has been extracted from the rest
+// of the user tables. This is in the hope of preventing anything stupid from
+// accidentally happening to it (log dumps, over-eager SELECT statements).
+type UserPII struct {
```
Comment (moogar0880) [inline users/pii.go:22]: Do we not want a `timestamp` field for UserPII?

File: users/auth.go [hunk 3]
```diff
@@ -1,54 +1,6 @@
 package users
```
Comment (moogar0880) [inline users/auth.go:1]: Just to make sure I follow, `users/auth` is more of just a utility package now?

Comment (jad-b) [inline users/auth.go:1]: Yeah, that's how it started shaking out. Once I finished re-arranging fields between UserAuth and UserPII, I ended up with the handful of auth-related functions that you now see in `auth.go`

File: users/pii.go (hunk 1 above)
Comment (jad-b) [inline users/pii.go]: Yeah, probably does not. I need to finish the switch.

File: users/pii.go (hunk 2 above)
Comment (jad-b) [inline users/pii.go:22]: Depends on what the timestamp is - LastModified? CreationTime?

**Comment by jad-b** [comment]
Opened this against jad-b/torque#basic-auth; going to re-open against master.

**Pull request closed without merging**